examples/Example/Page/Todos/TodoCSS.hs

```diff
@@ -0,0 +1,274 @@
+    Filter filt -> do
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+    SetCompleted filt todo completed -> do
+      _ <- Todos.setCompleted completed todo
+      todos <- filteredTodos filt
+      pure $ todosView filt todos
+   where
+    filteredTodos filt =
+      filter (isFilter filt) <$> Todos.loadAll
+
+    isFilter filt todo =
+      case filt of
+        FilterAll -> True
+        Active -> not todo.completed
+        Completed -> todo.completed
+
+todosView :: FilterTodo -> [Todo] -> View AllTodos ()
+todosView filt todos = do
+  header (extClass "header") $ do
+    h1 (style' "top:-80px") $ text "todos"
+    todoForm
+  main' (extClass "main") $ do
+    div' (extClass "toggle-all-container") $ do
+      input'
+        ( extClass "toggle-all"
+            . att "id" "toggle-all"
+            . att "type" "checkbox"
+        )
+      label'
+        ( extClass "toggle-all-label"
+            . att "for" "toggle-all"
+            . onClick (ToggleAll filt)
+        )
+        (text "Mark all as complete")
+    ul' (extClass "todo-list") $ do
+      forM_ todos $ \todo -> do
+        hyper (TodoView todo.id) $ todoView filt todo
+  statusBar filt todos
+
+todoForm :: View AllTodos ()
+todoForm = do
+  let f :: TodoForm FieldName = fieldNames
+  form SubmitTodo grow $ do
+    field f.task id $ do
+      Input (FieldName nm) <- context
+      input' -- we use a custom input field, because the Hyperbole one overrides autocomplete
+        ( extClass "new-todo"
+            {-
+              -- . autofocus
+              FIXME: turning off autofocus, that "steals" the focus on item click.
+              FIXME: to solve this, we could either store the "initially focused" state and track that boolean, or use buttons
+              FIXME: but since this example is meant to match as close as possible to the original CSS version
+              FIXME: and not diverge too much from the other todo example, I'm leaving as-is.
+             -}
+            . att "autocomplete" "off"
+            . placeholder "What needs to be done?"
+            . value ""
+            . name nm -- because we use a custom field, we must provide this param for the library
+        )
+
+statusBar :: FilterTodo -> [Todo] -> View AllTodos ()
+statusBar filt todos = do
+  footer (extClass "footer" . style' "padding-bottom: 30px") $ do
+    let numLeft = length $ filter (\t -> not t.completed) todos
+    span' (extClass "todo-count") $ do
+      text $
+        mconcat
+          [ T.pack $ show numLeft
+          , " "
+          , pluralize numLeft "item" "items"
+          , " "
+          , "left!"
+          ]
+    space
+    ul' (extClass "filters") $ do
+      filterLi FilterAll "All"
+      filterLi Active "Active"
+      filterLi Completed "Completed"
+    space
+    button ClearCompleted (extClass "clear-completed") "Clear completed"
+ where
+  filterLi f str =
+    li' (extClass "filter" . selectedFilter f) $ do
+      a
```

Comment:
Did you avoid using `Web.View.link` deliberately? Or just not know about it? `Web.View.Types.Url` has an `IsString` instance:

```
link "" (onClick (Filter f)) (text str)
```